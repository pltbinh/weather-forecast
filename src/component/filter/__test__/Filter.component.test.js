import React from 'react';
import renderer from 'react-test-renderer';
import FilterComponent from '../Filter.component';

// Snapshot testing
describe('FilterComponent', () => {
  it('should render correctly', () => {
    // act
    const component = renderer.create(<FilterComponent />);
    const tree = component.toJSON()

    // assert
    expect(tree).toMatchSnapshot()
  });

  it('should display loading state correctly', () => {
    // act
    const component = renderer.create(<FilterComponent loading />);
    const tree = component.toJSON()

    // assert
    expect(tree).toMatchSnapshot()
  });

  it('should display underline text correctly', () => {
    // act
    const component = renderer.create(<FilterComponent underlineText='some text' />);
    const tree = component.toJSON()

    // assert
    expect(tree).toMatchSnapshot()
  });
})