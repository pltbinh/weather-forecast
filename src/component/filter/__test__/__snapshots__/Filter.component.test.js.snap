// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterComponent should display loading state correctly 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="col-md-10 col-12"
    >
      <div
        className="form-group"
      >
        <input
          className="mb-3 form-control"
          id="searchText"
          onChange={[Function]}
          placeholder="Search here"
          type="text"
        />
      </div>
    </div>
    <div
      className="col-md-2 col-12"
    >
      <button
        className="btn btn-primary"
        disabled={true}
        id="btn-filter"
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <small
      className="text-muted mb-3 form-text"
    >
      
    </small>
  </div>
</form>
`;

exports[`FilterComponent should display underline text correctly 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="col-md-10 col-12"
    >
      <div
        className="form-group"
      >
        <input
          className="mb-3 form-control"
          id="searchText"
          onChange={[Function]}
          placeholder="Search here"
          type="text"
        />
      </div>
    </div>
    <div
      className="col-md-2 col-12"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        id="btn-filter"
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <small
      className="text-muted mb-3 form-text"
    >
      some text
    </small>
  </div>
</form>
`;

exports[`FilterComponent should render correctly 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="col-md-10 col-12"
    >
      <div
        className="form-group"
      >
        <input
          className="mb-3 form-control"
          id="searchText"
          onChange={[Function]}
          placeholder="Search here"
          type="text"
        />
      </div>
    </div>
    <div
      className="col-md-2 col-12"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        id="btn-filter"
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <small
      className="text-muted mb-3 form-text"
    >
      
    </small>
  </div>
</form>
`;

exports[`should display loading state correctly 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="col-md-10 col-12"
    >
      <div
        className="form-group"
      >
        <input
          className="mb-3 form-control"
          id="searchText"
          onChange={[Function]}
          placeholder="Search here"
          type="text"
        />
      </div>
    </div>
    <div
      className="col-md-2 col-12"
    >
      <button
        className="btn btn-primary"
        disabled={true}
        id="btn-filter"
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <small
      className="text-muted mb-3 form-text"
    >
      
    </small>
  </div>
</form>
`;

exports[`should display underline text correctly 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="col-md-10 col-12"
    >
      <div
        className="form-group"
      >
        <input
          className="mb-3 form-control"
          id="searchText"
          onChange={[Function]}
          placeholder="Search here"
          type="text"
        />
      </div>
    </div>
    <div
      className="col-md-2 col-12"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        id="btn-filter"
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <small
      className="text-muted mb-3 form-text"
    >
      some text
    </small>
  </div>
</form>
`;

exports[`should render correctly 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="col-md-10 col-12"
    >
      <div
        className="form-group"
      >
        <input
          className="mb-3 form-control"
          id="searchText"
          onChange={[Function]}
          placeholder="Search here"
          type="text"
        />
      </div>
    </div>
    <div
      className="col-md-2 col-12"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        id="btn-filter"
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <small
      className="text-muted mb-3 form-text"
    >
      
    </small>
  </div>
</form>
`;
